content/en/docs/advanced-tasks/catalog-update-formulary.md

```diff
@@ -0,0 +1,382 @@
+
+### semver template example
+Formulae presume the following content is saved to the file `semver.yaml`
+```yaml
+schema: olm.semver
+generatemajorchannels: false
+generateminorchannels: true
+candidate:
+  bundles:
+  - image: quay.io/organization/testoperator:v1.0.0
+  - image: quay.io/organization/testoperator:v1.0.1
+  - image: quay.io/organization/testoperator:v1.1.0
+fast:
+  bundles:
+  - image: quay.io/organization/testoperator:v1.0.1
+  - image: quay.io/organization/testoperator:v1.1.0
+stable:
+  bundles:
+  - image: quay.io/organization/testoperator:v1.0.1
+  ```
+
+
+### FBC example
+Formulae presume the following content is saved to the file `fbc.yaml`
+
+```yaml
+---
+defaultChannel: stable-v1.0
+name: testoperator
+schema: olm.package
+---
+entries:
+  - name: testoperator.v1.0.0
+  - name: testoperator.v1.0.1
+    skips:
+      - testoperator.v1.0.0
+name: candidate-v1.0
+package: testoperator
+schema: olm.channel
+---
+entries:
+  - name: testoperator.v1.1.0
+    replaces: testoperator.v1.0.1
+    skips:
+      - testoperator.v1.0.0
+name: candidate-v1.1
+package: testoperator
+schema: olm.channel
+---
+entries:
+  - name: testoperator.v1.0.1
+name: fast-v1.0
+package: testoperator
+schema: olm.channel
+---
+entries:
+  - name: testoperator.v1.1.0
+    replaces: testoperator.v1.0.1
+name: fast-v1.1
+package: testoperator
+schema: olm.channel
+---
+entries:
+  - name: testoperator.v1.0.1
+name: stable-v1.0
+package: testoperator
+schema: olm.channel
+```
+
+### rendered `olm.bundle` FBC example
+Formulae presume the following content is saved to the file `bundle.yaml`
+
+```yaml
+---
+image: quay.io/organization/testoperator:v1.1.0
+name: testoperator.v1.1.0
+package: testoperator
+properties:
+- type: olm.package
+  value:
+    packageName: testoperator
+    version: 1.1.0
+- type: olm.gvk
+  value:
+    group: api.testoperator.io
+    kind: Instance
+    version: v1alpha1
+relatedImages:
+- image: busybox
+  name: ""
+- image: quay.io/organization/testoperator:v1.1.0
+  name: ""
+schema: olm.bundle
+```
+
+## Formulae
+
+### Adding a new bundle to an existing channel
+#### semver
+Add a new `testoperatorv1.1.1` bundle pullspec to the `Candidate` channel archetype
+
+```bash
+yq eval '.candidate.bundles += [{"image" : "quay.io/organization/testoperator:v1.1.1"}]' semver.yaml
+```
+
+produces updated `Candidate` archetype contents:
+
+```yaml
+...
+candidate:
+  bundles:
+  - image: quay.io/organization/testoperator:v1.0.0
+  - image: quay.io/organization/testoperator:v1.0.1
+  - image: quay.io/organization/testoperator:v1.1.0
+  - image: quay.io/organization/testoperator:v1.1.1
+...
+```
+
+#### FBC
+Add a new `testoperator.v1.1.1` edge to an existing `candidate-v1.1` channel
+
+```bash
+yq eval 'select(.schema=="olm.channel" and .name == "candidate-v1.1").entries += [{"name" : "testoperator.v1.1.1"}]' fbc.yaml
```

Comment:
This is incorrect as it produces a second head.  The yq would be quite a bit more complicated.

The result should be:
```yaml
entries:
  - name: testoperator.v1.1.0
  - name: testoperator.v1.1.1
    replaces: testoperator.v1.1.0
    skips:
    - testoperator.v1.0.0
    - testoperator.v1.1.0
```

Reply:
I'll add a note that the FBC-side is incomplete as-is since it does not 'add a new head' but just 'add a new edge'. 
The real winner in that scenario is the semver template, where it is exactly that easy.